staging/src/k8s.io/apiserver/pkg/cel/environment/base.go

```diff
@@ -95,6 +94,34 @@ var baseOpts = []VersionedOptions{
 			ext.Strings(ext.StringsVersion(2)),
 		},
 	},
+
+	// add set ext library in 1.29
+	{
+		IntroducedVersion: version.MajorMinor(1, 29),
+		EnvOptions: []cel.EnvOption{
+			ext.Sets(),
+		},
+	},
+	// add the new validator in 1.29
+	{
+		IntroducedVersion: version.MajorMinor(1, 29),
+		EnvOptions: []cel.EnvOption{
+			cel.ASTValidators(
+				cel.ValidateDurationLiterals(),
+				cel.ValidateTimestampLiterals(),
+				cel.ValidateRegexLiterals(),
+				cel.ValidateHomogeneousAggregateLiterals(),
```

Comment:
Do we wanna enable `ValidateComprehensionNestingLimit` as well?

Reply:
You probably could, but it might be better to rely on the cost limits

Reply:
I agree. By focusing on cost limits, we only have one limit system to explain to users. If we accumulate data suggesting we need this in the future, we can reevaluate..